src/main/java/hw30/hibernate/StudentDao.java

```diff
@@ -0,0 +1,80 @@
+package hw30.hibernate;
+
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+public class StudentDao {
+
+    private final SessionFactory sessionFactory;
+
+    public StudentDao(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+    public void create() {
```

Comment:
Взагалі ДАО або репозиторій мають бути дуже прості і робити лише одну дію - працювати з БД. 
У тебе робить дні дії - створює ентіті і зберігає його. 
Правильніше, щоб студент приходив з зовні: 
```suggestion
    public void create(Student student) {
```

Reply:
Я не розумію, щодо репозиторію. Що ти маєш на увазі?

Reply:
Створити студента лише по імені - це вже бізнес логіка. Їй не місце в репозиторії чи дао. Вони мають робити лише одну задачу гарно - працювати з базою (зберігати, доставати). 
Для бізнес логіки є сервіс - це його задача створювати.